accounts-db/src/ancient_append_vecs.rs

```diff
@@ -91,15 +93,29 @@ impl AncientSlotInfos {
         can_randomly_shrink: bool,
         ideal_size: NonZeroU64,
         is_high_slot: bool,
+        db: &AccountsDb,
     ) -> bool {
+        let slots = [287302120, 287344734, 287344738, 287332411];
+
         let mut was_randomly_shrunk = false;
         let alive_bytes = storage.alive_bytes() as u64;
+        if slots.contains(&slot) {
+            log::error!(
+                "ancient.add? slot: {slot}, alive: {alive_bytes}, cap: {}, count: {}",
+                storage.capacity(),
+                storage.approx_stored_count()
+            );
+        }
         if alive_bytes > 0 {
             let capacity = storage.accounts.capacity();
             let should_shrink = if capacity > 0 {
-                let alive_ratio = alive_bytes * 100 / capacity;
-                alive_ratio < 90
-                    || if can_randomly_shrink && thread_rng().gen_range(0..10000) == 0 {
+                let candidate_for_shrink = db.is_candidate_for_shrink(&storage);
+                if candidate_for_shrink {
+                    // shrink should run on this
+                    db.shrink_candidate_slots.lock().unwrap().insert(slot);
+                }
+                candidate_for_shrink
+                    || if can_randomly_shrink && thread_rng().gen_range(0..80000) == 0 {
```

Comment:
this random constant needs to be proportional to total # ancient storages allowed.